pkg/registry/pod/etcd/etcd.go

```diff
@@ -321,7 +322,7 @@ func (r *AttachREST) Connect(ctx api.Context, name string, opts runtime.Object)
 	if err != nil {
 		return nil, err
 	}
-	return genericrest.NewUpgradeAwareProxyHandler(location, transport, true), nil
+	return genericrest.NewUpgradeAwareProxyHandler(location, transport, false, true), nil
```

Comment:
...should this use newUpgradeAwareProxyHandler? it's weird to have both in the file. If this is correct, a comment with an explanation is in order.